nmigen/lib/roundrobin.py

```diff
@@ -0,0 +1,54 @@
+from .. import *
+from functools import reduce
+from operator import or_
+
+
+__all__ = ["RoundRobin"]
+
+
+class RoundRobin(Elaboratable):
+    """Round-robin scheduler.
+
+    For a given set of requests, the round-robin scheduler will
+    grant one request.
+
+    Parameters
+    ----------
+    width : int
+        Number of requests.
+
+    Attributes
+    ----------
+    requests : Signal(width), in
+        Set of requests.
+    grant : Signal(range(width)), out
+        Number of the granted request.
+    valid : Signal(), out
+        Indicate if grant is valid.
```

Comment:
This comment should explain the conditions under which the grant is valid. (Right now it doesn't say anything the name doesn't already.)

Reply:
"Indicate if `grant` is a valid request that was emitted at the previous clock cycle."

That would be okay?